### PR TITLE
created a script to uninstall zabbix on head nodes

### PR DESCRIPTION
**Pull request by ekund**
This script will:
1. stop zabbix agent and servers
2. drop the zabbix table  from mysql
3. remove zabbix api ruby gems
4. remove all software and logs from the head nodes

### CODE REVIEW

**Comment by kiiranh** [comment]
@ekund You'll have to clear Zabbix [DB customization](https://github.com/bloomberg/chef-bach/blob/4768996a896beab1b93acc04e0f1c23980120988/cookbooks/bcpc/recipes/zabbix-head.rb#L121) and [Leader election](https://github.com/bloomberg/chef-bach/blob/4768996a896beab1b93acc04e0f1c23980120988/cookbooks/bcpc/recipes/zabbix-head.rb#L142) related files too.

**Comment by cbaenziger** [comment]
Please follow `rubocop` recommendations

**Comment by cbaenziger** [comment]
@kiiranh #441 covers the leader election issue but could you update that issue for tuning idempotency too; I suspect that has been missed for folks on upgrade?

@ekund, perhaps tackling #441 independently would be better than doing that here?

**Comment by ekund** [comment]
I have updated and cleaned up the script so that it passes rubocop, removed unneeded functions, and it can now work on both hardware and vm clusters.

File: uninstall_zabbix.rb [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  # This is really gross because Ruby 1.9 lacks Array#to_h.
+  File.readlines(File.join(repo_dir, 'cluster.txt'))
+    .map { |line| Hash[*fields.zip(line.split(' ')).flatten(1)] }
+end
+
+def repo_dir
+  File.dirname(__FILE__)
+end
+
+def check_status(c, on_fail_msg, on_success_msg, dofail = false)
```
Comment (cbaenziger) [inline uninstall_zabbix.rb]: Hi Esther, could you please provide a function comment here? (What is `c`?)

E.g.
````
##################
# check_status -- check status via parsing a c object
# expects: c - a c object
#              on_fail_msg - string
#              on_success_msg - string
#              dofail - to fail for fun and profit without checking something
# side affect: prints a message directly to the user via puts()

File: uninstall_zabbix.rb [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  end
+  mysqlcmd = 'mysql -uroot -p' + password + ' -e \'drop database zabbix;\''
+  c = Mixlib::ShellOut.new('./nodessh.sh', chef_env, vm_entry['ip_address'],
+                           mysqlcmd, 'sudo')
+  c.run_command
+  check_status(c, 'Could not drop zabbix database ', 'Dropped zabbix database',
+               true)
+end
+
+def uninstall_zabbix(chef_env)
```
Comment (cbaenziger) [inline uninstall_zabbix.rb]: Maybe for sanity's sake make a mention that this is to workaround Chef-BACH #441

File: uninstall_zabbix.rb (hunk 1 above)
Comment (cbaenziger) [inline uninstall_zabbix.rb]: It looks like c is a `Mixlib::ShellOut` object?

**Comment by cbaenziger** [comment]
Please squash commits and otherwise, things look good. The comment for `check_status` would be nice but is mainly a nit.

**Pull request merged**